pkg/cluster-operator/customcluster_helper.go

```diff
@@ -283,11 +283,21 @@ func generateClusterConfigKey(customCluster *v1alpha1.CustomCluster) client.Obje
 }
 
 func generateClusterHostsName(customCluster *v1alpha1.CustomCluster) string {
-	return customCluster.Name + "-" + ClusterHostsName
+	name := customCluster.Name + "-" + ClusterHostsName
+	name = strings.ToLower(name)
+	if len(name) > 63 {
+		name = name[:63]
```

Comment:
Two cents:

1. donot hardcode
2. If we strip, could we get duplicate resource names?